csrc/quantization/fp4/nvfp4_quant_kernels.cu

```diff
@@ -348,10 +348,7 @@ void scaled_fp4_quant_sm100a(torch::Tensor const& output,
   auto sf_out = static_cast<int32_t*>(output_sf.data_ptr());
   auto output_ptr = static_cast<int64_t*>(output.data_ptr());
   at::cuda::CUDAGuard device_guard{(char)input.get_device()};
-  auto stream = at::cuda::getStreamFromPool(false, input.get_device());
-  if (stream == nullptr) {
-    std::cerr << "Warning: Null CUDA stream" << std::endl;
-  }
+  auto stream = at::cuda::getCurrentCUDAStream(input.get_device());
```

Comment:
This looks good to me as it's more inline with other uses. What was the issue that popped up because of this?

Reply:
`getStreamFromPool` was giving a different stream for this kernel, so ops later on which are running on the main/current stream might try to access the result tensors before this kernel is finished. This was causing incorrect results and NaNs.

Reply:
We got some NaN values when the fp4 quant is issued to the wrong stream. I guess when we test e2e model. @trevor-m can provide more info.